Sources/SwiftGenKit/Parsers/StringsParser.swift

```diff
@@ -37,23 +40,48 @@ public final class StringsParser: Parser {
 
   // Localizable.strings files are generally UTF16, not UTF8!
   public func parse(path: Path) throws {
-    let name = path.lastComponentWithoutExtension
+    if path.isDirectory {
 
-    guard tables[name] == nil else {
-      throw StringsParserError.duplicateTable(name: name)
-    }
-    guard let data = try? path.read() else {
-      throw StringsParserError.failureOnLoading(path: path.string)
-    }
+      if try hasMultipleLprojSubdirectories(path: path) {
+        throw StringsParserError.multipleLprojDirectories
+      }
+
+      try _ = path.children().map { childPath in
+        if childPath.isDirectory || isStringsFile(name: childPath.lastComponent) {
+          try parse(path: childPath)
+        }
+      }
+    } else {
+      let name = path.lastComponentWithoutExtension
 
-    let plist = try PropertyListSerialization.propertyList(from: data, format: nil)
-    guard let dict = plist as? [String: String] else {
-      throw StringsParserError.invalidFormat
+      guard tables[name] == nil else {
+        throw StringsParserError.duplicateTable(name: name)
+      }
+      guard let data = try? path.read() else {
+        throw StringsParserError.failureOnLoading(path: path.string)
+      }
+
+      let plist = try PropertyListSerialization.propertyList(from: data, format: nil)
+      guard let dict = plist as? [String: String] else {
+        throw StringsParserError.invalidFormat
+      }
+
+      tables[name] = try dict.map { key, translation in
+        try Entry(key: key, translation: translation)
+      }
     }
+  }
 
-    tables[name] = try dict.map { key, translation in
-      try Entry(key: key, translation: translation)
+  private func hasMultipleLprojSubdirectories(path: Path) throws -> Bool {
+    var numberOfLprojDirectories = 0
+    for childPath in try path.children() {
+      numberOfLprojDirectories += childPath.lastComponent.hasSuffix(".lproj") ? 1 : 0
     }
```

Comment:
If you really wanted to count the total number of `.lproj` dirs you could have used `path.children.count(where: $0.lastComponent.hasSuffiz(".lproj")` directly instead of a for loop.

But:
1) It's not worth to continue counting as soon as you see more than one, so maybe instead of counting then check that count>1, you could exit the for loop early
2) this code doesn't seem to check subdirectories recursively. I can totally see people passing a path like `$PROJECT_DIR` by mistake but the localization files be in `$PROJECT_DIR/MyTarget/Resources/*.lproj` for example

Reply:
I did not intend to look for multiple `.lproj` subdirectories recursively, I intended to check for multiple `.lproj` subdirectories in one directory but I see your point.  
Just to be sure, you are suggesting to check recursively whether in the given `/path/to/directory` there are multiple `.lproj` subdirs so that the following example is "invalid"?  

`swiftgen strings -t structured-swift4 /path/to/directory`
```
- directory
  - pathA
    - en.lproj
  - pathB
    - es.lproj
```

Reply:
Mmmh that's a good question, as actually the use case you showed might be valid of pathA and pathB are directories corresponding to different targets (day different frameworks) in the same project…

(That's also why this would just be a warning and not an error, we want to prevent user mistakes but not block valid cases that we misinterpret as mistakes)

In your example though, even if pathA and pathB for example correspond to separate targets in the project then it's unlikely that the user will want to merge the keys of both targets into one giant constants output file, as some keys will be in one target but not the other anyway so they'll need two separate output files too…

So yeah I'd say in the case you presented we still might want to warn.

What would be even better / less error prone / more solid verification could be to check if we only have at most one language (e.g. having both `dir/pathA/en.lproj` and `dir/pathB/en.lproj` would be ok because overall we only have `en.lproj` so one language ; but your example mixing `en.lproj` and `es.lproj` is more likely to be a user mistake we could warn about

Reply:
Sorry for the extreeeeemely long delay :) ...I finally got to implementing the changes we discussed about